src/main/java/org/spongepowered/common/mixin/core/entity/item/MixinEntityEnderPearl.java

```diff
@@ -55,9 +59,14 @@ private boolean onEnderPearlImpact(EntityPlayerMP player) {
             return true;
         }
 
-        MoveEntityEvent.Teleport event = EntityUtil.handleDisplaceEntityTeleportEvent(player, this.getLocation());
-        if (event.isCancelled()) {
-            return true;
+        try (CauseStackManager.StackFrame frame = Sponge.getCauseStackManager().pushCauseFrame()) {
+            frame.addContext(EventContextKeys.TELEPORT_TYPE, TeleportTypes.ENTITY_TELEPORT);
+            frame.addContext(EventContextKeys.THROWER, (Player) player).getCurrentCause();
+
+            MoveEntityEvent.Teleport event = EntityUtil.handleDisplaceEntityTeleportEvent(player, this.getLocation());
```

Comment:
Cancelling the displace here, will also cancel the damage when it hits another entity. Not sure if this is intended.